core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -472,12 +471,11 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
   private[sql] def cachePutInto(doCache: Boolean, updateSubQuery: LogicalPlan,
       table: String): Option[LogicalPlan] = {
     // first acquire the global lock for putInto
-    val lock = SnappyContext.getClusterMode(sparkContext) match {
-      case _: ThinClientConnectorMode => null
-      case _ =>
-        PartitionedRegion.getRegionLock("PUTINTO_" + table, GemFireCacheImpl.getExisting)
-    }
-    if (lock ne null) lock.lock()
+    val (schemaName: String, _) =
+      JdbcExtendedUtils.getTableWithSchema(table, conn = null, Some(sqlContext.sparkSession))
+    val lock = grabLock(table, schemaName, defaultConnectionProps)
+
+    if (lock.isInstanceOf[RegionLock]) lock.asInstanceOf[RegionLock].lock()
```

Comment:
For uniformity I think you can move the lock acquisition inside grabLock method itself even for embedded mode. Cosmetic change, not insisting.